src/scmdata/run.py

```diff
@@ -1871,6 +1906,9 @@ def run_append(
 
     for run in runs[1:]:
         ret._ts.extend(run._ts)
+        for k, v in run.metadata.items():
```

Comment:
This block might be clearer and avoid looping unnecessarily if we did

```python
if metadata is None:
    for k, v in run.metadata.items():
        if k not in ret.metadata:
            ret.metadata[k] = v
```

Reply:
Moved the merging into a func instead. We can then swap/monkey_patch the functionality as needed